src/transformers/models/clip/modeling_tf_clip.py

```diff
@@ -0,0 +1,1528 @@
+        loss (:obj:`tf.Tensor` of shape :obj:`(1,)`, `optional`, returned when :obj:`return_loss` is :obj:`True`):
+            Contrastive loss for image-text similarity.
+        logits_per_image:(:obj:`tf.Tensor` of shape :obj:`(image_batch_size, text_batch_size)`):
+            The scaled dot product scores between :obj:`image_embeds` and :obj:`text_embeds`. This represents the
+            image-text similarity scores.
+        logits_per_text:(:obj:`tf.Tensor` of shape :obj:`(text_batch_size, image_batch_size)`):
+            The scaled dot product scores between :obj:`text_embeds` and :obj:`image_embeds`. This represents the
+            text-image similarity scores.
+        text_embeds(:obj:`tf.Tensor` of shape :obj:`(batch_size, output_dim`):
+            The text embeddings obtained by applying the projection layer to the pooled output of
+            :class:`~transformers.TFCLIPTextModel`.
+        image_embeds(:obj:`tf.Tensor` of shape :obj:`(batch_size, output_dim`):
+            The image embeddings obtained by applying the projection layer to the pooled output of
+            :class:`~transformers.TFCLIPVisionModel`.
+        text_model_output(:obj:`TFBaseModelOutputWithPooling`):
+            The output of the :class:`~transformers.TFCLIPTextModel`.
+        vision_model_output(:obj:`TFBaseModelOutputWithPooling`):
+            The output of the :class:`~transformers.TFCLIPVisionModel`.
+    """
+
+    loss: Optional[tf.Tensor] = None
+    logits_per_image: tf.Tensor = None
+    logits_per_text: tf.Tensor = None
+    text_embeds: tf.Tensor = None
+    image_embeds: tf.Tensor = None
+    text_model_output: TFBaseModelOutputWithPooling = None
+    vision_model_output: TFBaseModelOutputWithPooling = None
+
+    def to_tuple(self) -> Tuple[Any]:
+        return tuple(
+            self[k] if k not in ["text_model_output", "vision_model_output"] else getattr(self, k).to_tuple()
+            for k in self.keys()
+        )
+
+
+class TFCLIPVisionEmbeddings(tf.keras.layers.Layer):
+    def __init__(self, config: CLIPVisionConfig, **kwargs):
+        super().__init__(**kwargs)
+
+        self.embed_dim = config.hidden_size
+        self.image_size = config.image_size
+        self.patch_size = config.patch_size
+
+        self.num_patches = (self.image_size // self.patch_size) ** 2
+        self.num_positions = self.num_patches + 1
+
+        self.config = config
+
+        self.patch_embedding = tf.keras.layers.Conv2D(
+            filters=self.embed_dim,
+            kernel_size=self.patch_size,
+            strides=self.patch_size,
+            padding="valid",
+            data_format="channels_last",
+            use_bias=False,
+            kernel_initializer=get_initializer(self.config.initializer_range),
+            name="patch_embedding",
+        )
+
+    def build(self, input_shape: tf.TensorShape):
+
+        # Q: should we use normal initializer?
+        # initializer = tf.random_normal_initializer(mean=0.0, stddev=1.0)
```

Comment:
Needs to be addressed @patil-suraj do you have an answer?

Reply:
Aah yes! CLIP's initialization logic is bit complex, indeed we should use `normal` here.
See the ` CLIPPreTrainedModel.__init_weights` method for reference.

https://github.com/huggingface/transformers/blob/4cdb67caba9abee8b5b94b4f88f55e95a3d1013f/src/transformers/models/clip/modeling_clip.py#L345